### PR TITLE
Allows reloading from pistol belt.

### PR DESCRIPTION
**Pull request by hyper2snyper**
## About The Pull Request

Completes one of these approved suggestions.
https://tgstation13.org/phpBB/viewtopic.php?f=65&t=20487

Allows an empty pistol to be reloaded from a pistol belt using right click so long as right click is disabled and there is no ammo within.

## Why It's Good For The Game

https://tgstation13.org/phpBB/viewtopic.php?f=65&t=20487
It was an approved suggestion.
I think it would make reloading these low capacity weapons far easier. 

## Changelog
:cl:
add: Allows an empty pistol to be reloaded from a pistol belt using right click so long as right click is disabled and there is no ammo within.
/:cl:

### CODE REVIEW

File: code/game/objects/items/storage/belt.dm [hunk 1]
```diff
@@ -619,6 +620,37 @@
...
+
+/obj/item/storage/belt/gun/examine(mob/user, distance, infix, suffix)
+	. = ..()
+	to_chat(user, "<span class='notice'>Control-Click to toggle \'Tactical Reload Mode\'. While on, clicking on the belt with an empty pistol will perform a reload with the ammo inside.</span>")
+	if(!reload_on)
+		to_chat(user, "<span class='notice'>Reload mode is OFF</span>")
+	if(reload_on)
+		to_chat(user, "<span class='notice'>Reload mode is ON</span>")
+
+/obj/item/storage/belt/gun/CtrlClick(mob/user)
```
Comment (BraveMole) [inline code/game/objects/items/storage/belt.dm:645]: Maybe say on the examine that control click is the wait to change the reload mode?

File: code/game/objects/items/storage/belt.dm [hunk 2]
```diff
@@ -557,6 +557,7 @@
 		/obj/item/weapon/gun/pistol,
 		/obj/item/ammo_magazine/pistol,
 	)
+	var/reload_on = FALSE
```
Comment (BraveMole) [inline code/game/objects/items/storage/belt.dm:560]: autodoc

File: code/game/objects/items/storage/belt.dm [hunk 3]
```diff
@@ -619,6 +620,37 @@
 			return FALSE
 	return TRUE
 
+/obj/item/storage/belt/gun/attackby(obj/item/I, mob/user, params)
+	if(!istype(I, /obj/item/weapon/gun/pistol) || !reload_on)
+		return ..()
+	var/obj/item/weapon/gun/pistol/gun = I
+	if(gun.current_mag)
+		return ..()
```
Comment (BraveMole) [inline code/game/objects/items/storage/belt.dm:628]: probably want to to return
And a to chat to say that it has already a mag would be nice.
Also, if it's tactictal reload, should it not remove the mag already in place?

File: code/game/objects/items/storage/belt.dm [hunk 4]
```diff
@@ -619,6 +620,37 @@
...
+			gun.reload(user, mag)
+			orient2hud()
+			return
+
+/obj/item/storage/belt/gun/examine(mob/user, distance, infix, suffix)
+	. = ..()
+	to_chat(user, "<span class='notice'>Control-Click to toggle \'Tactical Reload Mode\'. While on, clicking on the belt with an empty pistol will perform a reload with the ammo inside.</span>")
+	if(!reload_on)
+		to_chat(user, "<span class='notice'>Reload mode is OFF</span>")
+	if(reload_on)
```
Comment (BraveMole) [inline code/game/objects/items/storage/belt.dm:642]: use an early return and remove the second if

File: code/game/objects/items/storage/belt.dm [hunk 5]
```diff
@@ -619,6 +620,37 @@
...
+		to_chat(user, "<span class='notice'>Reload mode is ON</span>")
+
+/obj/item/storage/belt/gun/CtrlClick(mob/user)
+	if(src.loc != user)
```
Comment (BraveMole) [inline code/game/objects/items/storage/belt.dm:646]: don't src.loc, just do src

File: code/game/objects/items/storage/belt.dm [hunk 6]
```diff
@@ -619,6 +620,37 @@
...
+		return ..()
+	reload_on = !reload_on
+	if(!reload_on)
+		to_chat(user, "<span class='notice'>You toggle the reload mode OFF.</span>")
+	if(reload_on)
```
Comment (BraveMole) [inline code/game/objects/items/storage/belt.dm:651]: early return

File: code/game/objects/items/storage/belt.dm (hunk 4 above)
Comment (TiviPlus) [inline code/game/objects/items/storage/belt.dm:642]: "reload mode" seems like a large hassle, and should likely be a different type of click, such as right or ctrl click onto the belt with a gun

**Review by TiviPlus — CHANGES_REQUESTED** [review]
Also @BraveMole use batch review pls

File: code/game/objects/items/storage/belt.dm (hunk 4 above)
Comment (hyper2snyper) [reply code/game/objects/items/storage/belt.dm:642]: I suppose so, but the suggestion did say to have a 
"reload mode." But I could always mess around a little to see what I can do.

File: code/game/objects/items/storage/belt.dm (hunk 1 above)
Comment (hyper2snyper) [reply code/game/objects/items/storage/belt.dm:645]: I did, in Examine()

File: code/game/objects/items/storage/belt.dm (hunk 3 above)
Comment (hyper2snyper) [reply code/game/objects/items/storage/belt.dm:628]: ah right, tactical means theres a mag. Ill change that.

File: code/game/objects/items/storage/belt.dm (hunk 5 above)
Comment (hyper2snyper) [reply code/game/objects/items/storage/belt.dm:646]: why? since ```src``` is the belt, and ```src.loc``` is the mob/tile where it is currently located. If I change it wouldn't ```src != user``` always equal true since ```/obj/item/storage/belt/gun``` doesn't ever equal ```/mob/user``` ?

File: code/game/objects/items/storage/belt.dm [hunk 7]
```diff
@@ -619,15 +619,40 @@
...
+	var/obj/item/weapon/gun/pistol/gun = I
+	if(gun.current_mag)
+		return ..()
+	for(var/obj/item/ammo_magazine/mag in contents) 
+		if(!istype(gun, mag.gun_type))
+			continue
+		else
+			gun.reload(user, mag)
+			orient2hud()
+			return
```
Comment (TiviPlus) [inline code/game/objects/items/storage/belt.dm:641]:
```suggestion
		gun.reload(user, mag)
		orient2hud()
		return
```

File: code/game/objects/items/storage/belt.dm [hunk 8]
```diff
@@ -619,15 +619,40 @@
...
+		return ..()
+	var/obj/item/weapon/gun/pistol/gun = I
+	if(gun.current_mag)
+		return ..()
+	for(var/obj/item/ammo_magazine/mag in contents) 
+		if(!istype(gun, mag.gun_type))
+			continue
+		else
+			gun.reload(user, mag)
+			orient2hud()
```
Comment (TiviPlus) [inline code/game/objects/items/storage/belt.dm:640]: Why is this needed?

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/game/objects/items/storage/belt.dm (hunk 7 above)
Comment (hyper2snyper) [reply code/game/objects/items/storage/belt.dm:641]: its not

File: code/game/objects/items/storage/belt.dm (hunk 8 above)
Comment (hyper2snyper) [reply code/game/objects/items/storage/belt.dm]: its not, thanks

**Review by TiviPlus — APPROVED** [review]

**Comment by Dolth** [comment]
Honestly I'd like to know who came up with this idea...
